lib/ibm_watson/watson_service.rb

```diff
@@ -125,14 +118,14 @@ def request(args)
     args.delete_if { |_, v| v.nil? }
     args[:headers].delete("Content-Type") if args.key?(:form) || args[:json].nil?
 
-    if @username == "apikey"
-      _iam_apikey(iam_apikey: @password)
+    if @username == "apikey" && !@icp_prefix
+      iam_apikey(iam_apikey: @password)
```

Comment:
Do we need an `else` here to authenticate using basicauth? I see we do it in `initialize`. Do we need it in `request` as well?

Reply:
@mediumTaj  I believe that this is included for if a user passes in `"apikey"` as a username after the object is instantiated. A couple lines after this, the method determines which authentication to use. Since the `iam_apikey` was recently passed in using the username/password method, it needs to instantiate the token manager.

Reply:
Basic auth is at line [131](https://github.com/watson-developer-cloud/ruby-sdk/pull/27/files/1108559d78dbf7b5001422924bd42ecaadfbcc77#diff-619006c343086f2404542ab608f3a98eR131)

It is technically just a repetition, the only difference being that we are making `username` nill here. And I think we can get rid of the repetition. I will update.

Reply:
ah my bad, i didnt notice the basicauth down there!

Reply:
Hi @maxnussbaum !